app/upgrades/v19/upgrades.go

```diff
@@ -0,0 +1,46 @@
+package v19
+
+import (
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+)
+
+const (
+	UpgradeName = "v19"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v19
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	cdc codec.Codec,
+	wasmKeeper wasmkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade v19...")
+
+		// Run module migrations first to add wasm to the store
+		ctx.Logger().Info("Running module migrations...")
+		newVm, err := mm.RunMigrations(ctx, configurator, vm)
+		if err != nil {
+			return newVm, err
+		}
+
+		// Update wasm params so that contracts can only be uploaded through governance
+		wasmParams := wasmKeeper.GetParams(ctx)
+		wasmParams.CodeUploadAccess = wasmtypes.AccessConfig{
+			Permission: wasmtypes.AccessTypeNobody,
```

Comment:
So this means when we deploy to the real main net we will deploy hyperlane via gov props?

Reply:
yeah that's the current plan. I think there's a chance we may want to switch it to an address but we can handle that in a separate PR if needed